### DISCUSSION TITLE
Running `sudo stratis pool create platter /dev/sda` results in the mkfs.xfs error "Filesystem must be larger than 300MB"

### DISCUSSION THREAD

**Discussion by stuartpb** [stratis-storage/project#524] [Q&A] [answered]
This is on a fresh (set up a few days ago) installation of Arch Linux. I'm trying to set up a new Stratis pool on a 2TB USB hard drive:

```
[stuart@studlow ~]$ sudo stratis pool create platter /dev/sda 
Execution failed:
stratisd failed to perform the operation that you requested. It returned the following information via the D-Bus: ERROR: Error: Command failed: cmd: "/usr/sbin/mkfs.xfs" "-f" "-q" "/dev/dm-3" "-m" "uuid=2a6ec9d7-0679-4b90-a0fa-b4179894b83b" "-d" "noalign", exit reason: 1 stdout:  stderr: Filesystem must be larger than 300MB.
```

Trying to rerun the mkfs.xfs command manually results in the same error message - it appears that `/dev/dm-3` points to a partition that's only 16MB. Some diagnostic info:

```
[stuart@studlow ~]$ lsblk
NAME                                                                    MAJ:MIN RM   SIZE RO TYPE    MOUNTPOINTS
sda                                                                       8:0    0   1.8T  0 disk    
└─stratis-1-private-2a6ec9d706794b90a0fab4179894b83b-physical-originsub 254:0    0   816M  0 stratis 
  ├─stratis-1-private-2a6ec9d706794b90a0fab4179894b83b-flex-thinmeta    254:1    0    16M  0 stratis 
  ├─stratis-1-private-2a6ec9d706794b90a0fab4179894b83b-flex-thindata    254:2    0   768M  0 stratis 
  └─stratis-1-private-2a6ec9d706794b90a0fab4179894b83b-flex-mdv         254:3    0    16M  0 stratis 
[stuart@studlow ~]$ sudo dmsetup ls
stratis-1-private-2a6ec9d706794b90a0fab4179894b83b-flex-mdv     (254:3)
stratis-1-private-2a6ec9d706794b90a0fab4179894b83b-flex-thindata        (254:2)
stratis-1-private-2a6ec9d706794b90a0fab4179894b83b-flex-thinmeta        (254:1)
stratis-1-private-2a6ec9d706794b90a0fab4179894b83b-physical-originsub   (254:0)
```

**Comment by bmr-cymru** [comment, marked as answer]
The XFS project has changed the minimum allowed size of an xfs file system in recent versions of the userspace tools. In older versions of Stratis the default size for the metadata volume (MDV) was 16MiB as you see above - this is no longer permitted by current xfsprogs and results in the error you report.

Stratis was updated to use 512MiB as the initial MDV size in commit stratis-storage/stratisd@6bc82737260e0719e074ff3fba30e5095d76c998  - included in [stratisd-3.1](https://github.com/stratis-storage/stratisd/releases/tag/v3.1.0) onwards.

Looking at the Arch website it seems that they are [currently shipping stratisd-2.4.4](https://archlinux.org/packages/?sort=&q=stratis&maintainer=&flagged=) which does not include this change.

You would need to update to a more recent version of stratisd either by building from source or by filing a request to update the Arch packages. The current release of stratisd is [3.2.3](https://github.com/stratis-storage/stratisd/releases/tag/v3.2.3) and this includes additional fixes and features compared to 3.1. The CLI tools are currently on version [3.2.0](https://github.com/stratis-storage/stratisd/releases/tag/v3.2.3) and we would recommend updating to a newer CLI version when updating the daemon.

**Comment by stuartpb** [comment]
Link to the 2022-08-04 upstream change: https://git.kernel.org/pub/scm/fs/xfs/xfsprogs-dev.git/commit/?id=6e0ed3d19c54603f0f7d628ea04b550151d8a262

**Comment by stuartpb** [comment]
Downstream ticket: https://bugs.archlinux.org/task/76080
